plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -1289,8 +1292,7 @@ controller& chain_plugin::chain() { return *my->chain; }
 const controller& chain_plugin::chain() const { return *my->chain; }
 
 chain::chain_id_type chain_plugin::get_chain_id()const {
-   EOS_ASSERT( my->chain_id.valid(), chain_id_type_exception, "chain ID has not been initialized yet" );
-   return *my->chain_id;
+   return my->chain->get_chain_id();
```

Comment:
Add ` EOS_ASSERT( my->chain, chain_id_type_exception, "chain has not been initialized yet" );` to protect against undefined behavior in case someone calls `get_chain_id` before chain has been initialized.